src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -2469,10 +2474,19 @@ module Crystal
       node.exp.accept self
       @in_type_args -= 1
 
+      type = node.exp.type?
+      unless type
+        raise "BUG: #{node} at #{node.location} should receive a type"
+      end
+
+      if type.is_a? GenericClassType
```

Comment:
This should be `GenericType`. That's the base module for both generic classes and modules.